src/cluster/placement/algo/sharded_prop_test.go

```diff
@@ -0,0 +1,89 @@
+package algo
+
+import (
+	"fmt"
+	"math"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/leanovate/gopter"
+	"github.com/leanovate/gopter/gen"
+	"github.com/leanovate/gopter/prop"
+
+	"github.com/m3db/m3/src/cluster/placement"
+)
+
+const minSuccessfulTests = 100
+
+func TestInitialPlacementIsBalancedPropTest(t *testing.T) {
+	var (
+		parameters = gopter.DefaultTestParameters()
+		seed       = time.Now().UnixNano()
+		props      = gopter.NewProperties(parameters)
+		reporter   = gopter.NewFormatedReporter(true, 160, os.Stdout)
+	)
+
+	parameters.MinSuccessfulTests = minSuccessfulTests
+	parameters.Rng.Seed(seed)
+
+	props.Property("Initial placement is balanced", prop.ForAll(
+		testInitialPlacementIsBalanced,
+		gen.IntRange(1, 3),
+		gen.IntRange(1, 20),
+		gen.IntRange(128, 1024),
```

Comment:
Let's cover a broader shard range based on what we've seen (maybe 64 to 3072).

Reply:
https://github.com/m3db/m3/pull/4020/commits/49585eeb2a14216bc3cbec71d073e39b634ae98d